aiidalab_widgets_base/structures.py

```diff
@@ -85,7 +84,7 @@ def __init__(
             self.viewer = viewer
         else:
             self.viewer = StructureDataViewer(**kwargs)
-        tl.dlink((self, "structure_node"), (self.viewer, "structure"))
+        tl.dlink((self, "structure"), (self.viewer, "structure"))
```

Comment:
It was a bug to link `structure_node`. The `structure` trait is supposed to contain the current state of things, while `structure_node` is an unstored AiiDA object.

In the previous (buggy) version the AiiDA node would then be converted back to an ASE object losing loads of information.

Reply:
Maybe it is a good idea to make the `structure_node` trait immutable because it is not meant to be modified externally. Its job is literally to be the StructureNode object corresponding to the `structure` trait (ASE).

> Hmm, but I think there were some design decisions involved in that.

And, again, it was a mistake (probably mine) to link the `structure_node` trait instead of the `structure`.

Reply:
> Maybe it is a good idea to make the structure_node trait immutable because it is not meant to be modified externally.

`structure_node` trait is already read-only so I think we're good here. Thanks for the explanation. I'll leave this unresolved just to keep this discussion visible.